From 246619b69ba80675603587ae6c18686c37359880 Mon Sep 17 00:00:00 2001
From: Viorel Suman <viorel.suman@nxp.com>
Date: Fri, 2 Feb 2018 14:52:04 +0200
Subject: [PATCH 3409/5242] MLK-17516: ASoC: fsl: amix: refine the error
 handling

commit  bf50cdb80416d3451810fe6748397f68f1fbe8be from
https://source.codeaurora.org/external/imx/linux-imx.git

The "snd_pcm_hw_constraint_minmax" call may return a positive
non-error integer so that the subsequent "snd_pcm_hw_constraint_mask64"
call is never invoked, thus the formats are never enforced.
Fix the error handling so that only negative results are considered.

Signed-off-by: Viorel Suman <viorel.suman@nxp.com>
Reviewed-by: Daniel Baluta <daniel.baluta@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/imx-amix.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sound/soc/fsl/imx-amix.c b/sound/soc/fsl/imx-amix.c
index bd64e6b..d22ff25 100644
--- a/sound/soc/fsl/imx-amix.c
+++ b/sound/soc/fsl/imx-amix.c
@@ -53,14 +53,14 @@ static int imx_amix_fe_startup(struct snd_pcm_substream *substream)
 	if (clk_rate % 24576000 == 0) {
 		ret = snd_pcm_hw_constraint_list(runtime, 0,
 			SNDRV_PCM_HW_PARAM_RATE, &imx_amix_rate_constraints);
-		if (ret)
+		if (ret < 0)
 			return ret;
 	} else
 		dev_warn(dev, "mclk may be not supported %lu\n", clk_rate);
 
 	ret = snd_pcm_hw_constraint_minmax(runtime,
 			SNDRV_PCM_HW_PARAM_CHANNELS, 1, 8);
-	if (ret)
+	if (ret < 0)
 		return ret;
 
 	return snd_pcm_hw_constraint_mask64(runtime,
-- 
1.7.9.5

